onyo/commands/edit.py

```diff
@@ -47,6 +87,7 @@ def edit(args, onyo_root):
     problems are found, the choice will be offered to reopen the editor to fix
     them, or abort and return to the original state.
     """
+    repo = ""
```

Comment:
Pyre thinks different, because it does not know about `sys.exit(1)` in catch and thinks repo would not be initialized in the error case:
`Uninitialized local [61]: Local variable `repo` is undefined, or not always defined`
(https://github.com/psyinfra/onyo/actions/runs/3291764861/jobs/5426323059 shows me trying to push it without the line.)

https://pyre-check.org/docs/errors/ has a section about initilizing in try/catch specifically, which says one should either initilize before, or alternatively if that is not liked, move all the following code that uses repo into the try block, which is considered bad practice.
Repo has no empty initilizer, so I choose the string.
I will check for different solutions again later, but this is why this is right now unfixed.

Reply:
Then `repo = None` would be more appropriate.